tests/uv/manager/test_measurementmanager.py

```diff
@@ -0,0 +1,248 @@
+    test_manager.process(i, {})
+
+  # check data_store
+  assert data_store == {
+      'Int_10': [{
+          'step': 0,
+          'value': 0
+      }, {
+          'step': 10,
+          'value': 0
+      }, {
+          'step': 20,
+          'value': 0
+      }, {
+          'step': 30,
+          'value': 0
+      }, {
+          'step': 40,
+          'value': 0
+      }, {
+          'step': 50,
+          'value': 0
+      }, {
+          'step': 60,
+          'value': 0
+      }, {
+          'step': 70,
+          'value': 0
+      }, {
+          'step': 80,
+          'value': 0
+      }, {
+          'step': 90,
+          'value': 0
+      }],
+      'Int_13': [{
+          'step': 0,
+          'value': 0
+      }, {
+          'step': 13,
+          'value': 0
+      }, {
+          'step': 26,
+          'value': 0
+      }, {
+          'step': 39,
+          'value': 0
+      }, {
+          'step': 52,
+          'value': 0
+      }, {
+          'step': 65,
+          'value': 0
+      }, {
+          'step': 78,
+          'value': 0
+      }, {
+          'step': 91,
+          'value': 0
+      }],
+      'Int_17': [{
+          'step': 0,
+          'value': 0
+      }, {
+          'step': 17,
+          'value': 0
+      }, {
+          'step': 34,
+          'value': 0
+      }, {
+          'step': 51,
+          'value': 0
+      }, {
+          'step': 68,
+          'value': 0
+      }, {
+          'step': 85,
+          'value': 0
+      }],
+      'Int_20': [{
+          'step': 0,
+          'value': 0
+      }, {
+          'step': 20,
+          'value': 0
+      }, {
+          'step': 40,
+          'value': 0
+      }, {
+          'step': 60,
+          'value': 0
+      }, {
+          'step': 80,
+          'value': 0
+      }]
+  }
+
+
+def test_state_usage():
+  """ Tests that the measurements are made on the proper state,
+  i.e. the static and dynamic states are used appropriately """
+  data_store = {}
+  reporter = r.MemoryReporter(data_store).stepped()
+  static_state = {'Value1': 2., 'Value2': 4.}
+
+  test_manager = m.MeasurementManager(static_state, reporter)
+
+  test_manager.add_measurement({
+      'name': 'Value1',
+      'interval': 3,
+      'function': lambda x: x['Value1']
+  })
+  test_manager.add_measurement({
+      'name': 'Value2',
+      'interval': 3,
+      'function': lambda x: x['Value2']
+  })
+  test_manager.add_measurement({
+      'name': 'sum_12',
+      'interval': 3,
+      'function': lambda x: x['Value2'] + x['Value1']
+  })
+  test_manager.add_measurement({
+      'name': 'sum_13',
+      'interval': 3,
+      'function': lambda x: x['Value1'] + x['Value3']
+  })
+  test_manager.add_measurement({
+      'name': 'Value3',
+      'interval': 3,
+      'function': lambda x: x['Value3']
+  })
+
+  TEST_STEPS = 5
+  for step in range(TEST_STEPS):
+    dynamic_state = {'Value3': step * step}
+    test_manager.process(step, dynamic_state)
+
+  assert data_store == {
```

Comment:
same here, mind tightening up how we generate this list?

Reply:
Yep, done!